python/lsst/daf/butler/core/datastore.py

```
@@ -277,23 +288,23 @@ def __init__(self, config, bridgeManager, butlerRoot=None):
         constraintsConfig = self.config.get("constraints")
         self.constraints = Constraints(constraintsConfig, universe=bridgeManager.universe)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.name
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return self.name
 
     @property
-    def names(self):
+    def names(self) -> List[str]:
```

Returning a `list` via a read-only property might be confusing - it's possible to write something like `obj.names[0] = "zero"`, and have it silently do nothing.  Could you return a `tuple` here instead?

---

That is a good point. I should be returning a tuple.